### DISCUSSION TITLE
Auto Uppercase Sluglines and Characters

### DISCUSSION THREAD

**Discussion by moderidden** [chuangcaleb/obsidian-fountain-editor#35] [Ideas]
Thank you for making this plugin! I don't know if it's possible, but could you have any line that begins with "INT." or "EXT." automatically UPPERCASE on pressing enter. Similarly, could any line that doesn't end with a period also auto UPPERCASE (i.e. a character name). Those first two would be my biggest feature requests, but if I had a pie in the sky one, it'd be automatically creating a character list and have it auto suggest after two or three keystrokes.

**Comment by chuangcaleb** [comment]
Hello! Thank you for reaching out! Just hearing your thanks has made my day.

## Auto-Uppercase [Scene Headings](https://fountain.io/syntax/#scene-headings)

Regarding sluglines auto-UPPERCASING, yeah that sounds good and foolproof to implement. We can add that to the Project board.

To do that for any line that doesn't end with a period may be trickier. Some lines may end with other punctuation. Or maybe lines that don't end in punctuation? How would this work?

## Character Name Autocomplete

But a character name autocomplete!!!!! That's QoL. We can use the `@` symbol to trigger an autocomplete modal. 

One way: we can populate the list from a list object in the file's frontmatter/metadata/properties. But I have my sights set on larger projects that may use a multi-file structure (that Obsidian can take advantage of), so there could be a special index file at the root of a project folder, like how the Longform plugin does it.

Another idea I had a while ago was to use `[[Links]]` to the corresponding note of the character. So you'd have a note for each character, and then you'd use `[[CALEB]]` as the Character token. Two amazing things about this: one, if you ever rename your character, then every instance in your script gets renamed too. Next, you get (back)links connecting your character note to each Fountain screenplay note they are in.

## Time

Right now, I do not have the resource allocation to develop these extra features into these project right now. But once I do, these certainly will be implemented first because these are really good ideas. Or perhaps if a community member would like to make a contribution to add this to the project. Anyways, thank you for your input!

**Reply by moderidden** [reply]
Thanks for your reply! I am currently using the Compltr plugin which is connected to a word list Note in obsidian that has every character name in all caps on a new line. It works well for me for the characters, its just annoying to have to enable and disable it when I'm working on a screenplay since I don't want the autocomplete popping up when I don't want to.

It would be cool to integrate the character list into an Index file, which auto populated itself everytime a character name was used, or even just manually entering a character list in the frontmatter, though that might get a little convoluted.

But yes, to address your quesiton, I meant more that any line that didn't end in punctuation would be assumed to be a character line, especially if it was followed by a line directly below it, i.e.:

character name
This is the second line.

would reformat roughly to:

               CHARACTER NAME
        This is the second line.

As for the sluglines, that would be great! For me, I am opening up the .md files in Highland Pro to export to a PDF, and it automatically capitalizes sluglines there, but it would be nice as I'm going along to see that, especially if it's an easy fix.

Thanks again for your work on this plugin!

**Comment by chuangcaleb** [comment]
Just revisiting the idea of Capitalised Scene Headings/Sluglines, may need to spec this feature out in this thread.

Are there cases that we also would want a lowercase letter in the Scene Heading? e.g. `EXT. O'McDONALD'S FARM`
Not sure if there are any ases

**Reply by moderidden** [reply]
To my knowledge, there are no cases where a lower case letter would be used in a slugline. But maybe there would be a way to build in an override, such as starting a line with a certain punctuation would ignore auto formatting.

thanks for revisiting this!

**Reply by chuangcaleb** [reply]
yeah i can't think of any cases either.

I am trying to keep the plugin's philosophy of being a "renderer" and not add complexity with possibly destructive updates. maybe could lock this behind a default-disabled setting. I'll add this to the board!

**Comment by chuangcaleb** [comment]
Regarding a shortcut for auto-suggesting character names, I just stumbled across this plugin, this seems to be exactly it. I may try it myself soon!
[GitHub - saibotsivad/obsidian-at-people: Obsidian plugin to use the familiar @ notation to cross link to people.](https://github.com/saibotsivad/obsidian-at-people)
